### PR TITLE
(fix) O3-2666: Order getting removed after clicking on Back to order basket

### PR DESCRIPTION
**Pull request by usamaidrsk**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary

- Fix Order getting removed after clicking on `Back to order basket`

## Related Issue

[O3-2666](https://openmrs.atlassian.net/browse/O3-2666)

### CODE REVIEW

File: packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx [hunk 1]
```diff
@@ -147,17 +147,6 @@ describe('AddDrugOrderWorkspace drug search', () => {
...
-      expect.objectContaining({
-        ...getTemplateOrderBasketItem(
-          mockDrugSearchResultApiData[0],
-          undefined,
-          mockDrugOrderTemplateApiData[mockDrugSearchResultApiData[0].uuid][0],
-        ),
-        startDate: expect.any(Date),
-      }),
-    ]);
-
```
Comment (vasharma05) [inline packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx:160]: Please update the test, we should never remove a test, we should modify the test file.

File: packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx [hunk 2]
```diff
@@ -172,16 +161,6 @@ describe('AddDrugOrderWorkspace drug search', () => {
...
-    expect(hookResult.current.orders).toEqual([
-      expect.objectContaining({
-        ...getTemplateOrderBasketItem(
-          mockDrugSearchResultApiData[0],
-          undefined,
-          mockDrugOrderTemplateApiData[mockDrugSearchResultApiData[0].uuid][0],
-        ),
-        startDate: expect.any(Date),
-      }),
-    ]);
```
Comment (vasharma05) [inline packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx:184]: Same here

File: packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx (hunk 1 above)
Comment (usamaidrsk) [reply packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx:160]: Thanks, the test I removed was checking if this order was saved in orders store, which is not  the case when user visits order form and gets back to the basket, given the new requirements

CC @vasharma05

File: packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx (hunk 2 above)
Comment (usamaidrsk) [reply packages/esm-patient-medications-app/src/add-drug-order/add-drug-order.test.tsx:184]: https://github.com/openmrs/openmrs-esm-patient-chart/pull/1549#discussion_r1431010787

**Comment by brandones** [comment]
Hi Usama! Thanks for the PR. A note on how we handle PRs.

image

The checkboxes are there both to remind the submitter to do stuff and to communicate with others. You should not check them if they are not relevant. The one about conforming to the styleguide informs the reviewer that you have done some design implementation that references some specific designs. That isn't relevant to this ticket, so don't check the box. The one about tests requires some careful thought; the question is whether tests would fail if you didn't make the change you made. Tests were passing before (with the bad behavior), and you have only removed tests; therefore it is impossible that if your change was somehow undone that tests would fail. Therefore your change is not tested for, and you shouldn't check the box. It's not strictly a requirement, but if it's not too hard to write a test that protects us against this bug, it would be great to have.

**Comment by usamaidrsk** [comment]
> Hi Usama! Thanks for the PR. A note on how we handle PRs.
> 
> image
> 
> The checkboxes are there both to remind the submitter to do stuff and to communicate with others. You should not check them if they are not relevant. The one about conforming to the styleguide informs the reviewer that you have done some design implementation that references some specific designs. That isn't relevant to this ticket, so don't check the box. The one about tests requires some careful thought; the question is whether tests would fail if you didn't make the change you made. Tests were passing before (with the bad behavior), and you have only removed tests; therefore it is impossible that if your change was somehow undone that tests would fail. Therefore your change is not tested for, and you shouldn't check the box. It's not strictly a requirement, but if it's not too hard to write a test that protects us against this bug, it would be great to have.

Thanks @brandones, It's a good guide, and I will endeavor to follow along. Also with the tests I am to add the tests as to the changes I made.

**Review by brandones — APPROVED** [review]
LGTM, thanks @usamaidrsk & @vasharma05

**Pull request merged**